src/test/java/net/greatstart/controllers/InvestmentControllerTest.java

```diff
@@ -0,0 +1,184 @@
+
+    @Mock
+    private InvestmentValidationService investmentValidationService;
+    @Mock
+    private Principal principal;
+    @Mock
+    private InvestmentService investmentService;
+    @Mock
+    private ProjectService projectService;
+    @Mock
+    private UserService userService;
+    @InjectMocks
+    private InvestmentController investmentController;
+    private Project project;
+    private List<Investment> investments;
+    private MockMvc mvc;
+    @Captor
+    private ArgumentCaptor<Investment> captor;
+
+    @Before
+    public void setUp() throws Exception {
+        mvc = standaloneSetup(investmentController).build();
+        project = new Project();
+        Investment investment1 = new Investment();
+        investment1.setSum(new BigDecimal(3000));
+        Investment investment2 = new Investment();
+        investment2.setSum(new BigDecimal(2000));
+        investments = Arrays.asList(investment1, investment2);
+        project.setInvestments(investments);
+    }
+
+    @Test(timeout = 2000)
+    public void getAddInvestmentForm() throws Exception {
+        when(projectService.getProjectById(1)).thenReturn(project);
+        mvc.perform(get("/project/1/investment/add"))
+                .andExpect(view().name("investment/add_investment"))
+                .andExpect(model().attribute("project", project))
+                .andExpect(model().attribute("investedSum", new BigDecimal(5000)));
+    }
+
+    @Test(timeout = 2000)
+    public void addCorrectInvestment() throws Exception {
+        User user = new User();
+        user.setName("Ivan");
+        user.setLastName("Mazepa");
+        user.setEmail(TEST_EMAIL);
+        Investment investment = createInvestment(project, user, new BigDecimal(2000));
+        when(principal.getName()).thenReturn(TEST_EMAIL);
+        when(userService.getUserByEmail(TEST_EMAIL)).thenReturn(user);
+        when(projectService.getProjectById(1)).thenReturn(project);
+        when(investmentValidationService.validate(new BigDecimal(2000), project))
+                .thenReturn(null);
+        mvc.perform(post("/project/1/investment/add")
+                .principal(principal)
+                .param("sum", "2000"))
+                .andExpect(view().name(REDIRECT + PROJECT_PAGE + 1));
+
+        verify(investmentService, times(1)).saveInvestment(captor.capture());
+        assertEquals(investment.getSum(), captor.getValue().getSum());
+        assertEquals(investment.getInvestor(), captor.getValue().getInvestor());
+        assertEquals(investment.getProject(), captor.getValue().getProject());
+        assertTrue((investment.getDateOfInvestment()
+                .compareTo(captor.getValue().getDateOfInvestment()) <= 1)
+                && (investment.getDateOfInvestment()
+                .compareTo(captor.getValue().getDateOfInvestment()) >= -1));
+    }
+
+    @Test(timeout = 2000)
+    public void addCWrongInvestment() throws Exception {
+        User user = new User();
```

Comment:
You made this at least in 2 tests. Why not move this user to an instance variable and then just use it in several tests?